.husky/pre-commit

```diff
@@ -0,0 +1,21 @@
+#npm run text 
+
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+# Run TypeScript compiler in noEmit mode to check for errors in client and server
+npx --prefix ./client tsc --noEmit --project ./client/tsconfig.json
+CLIENT_TSC=$?
+
+npx --prefix ./server tsc --noEmit --project ./server/tsconfig.json
+SERVER_TSC=$?
+
+if [ $CLIENT_TSC -ne 0 ] || [ $SERVER_TSC -ne 0 ]; then
+  echo "TypeScript errors detected. Commit aborted."
+  exit 1
+fi
+
+# If tsc passes, then run lint-staged
+cd client && npm run lint && npm run format
+cd ..
+cd server && npm run lint && npm run format
```

Comment:
I'm noticing a couple issues here:

### 1. Command order
 
When a prettier warning has been introduced to the code, it seem like `npm run lint` exits with a status code of 1 and `npm run format` is not subsequently run. I think this stems from the `&&` operator.

I tested this by adding some echo commands:
```
# cd client && npm run lint && npm run format
cd client && echo 'RUN LINT' && npm run lint && echo 'RUN FORMAT' && npm run format
echo 'MOVING TO ROOT DIR'
```

In the output, you won't see `RUN FORMAT` before seeing `moving to root dir`:
```
RUN LINT

> linkta@1.0.0 lint
> eslint . && prettier --check .


Core/client/components/layout/MainLayout.tsx
  39:50  warning  Replace `"layout-open"` with `'layout-open'`  prettier/prettier

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

Checking formatting...
[warn] components/layout/MainLayout.tsx
[warn] Code style issues found in the above file. Run Prettier with --write to fix.
MOVING TO ROOT DIR
```

This means that what's happening in the `npm run lint` command is preventing whatever follows `&&` from occurring. I thought maybe whatever follows `&&` won't run if a process exits with a non-zero status code. Running `npm run lint` in a subshell confirms the exit status code, but also ends up allowing the format script to run:
```
cd client && echo 'RUN LINT' && (npm run lint; LINT_STATUS=$?; echo "npm run lint exited with status code $LINT_STATUS") && echo 'RUN FORMAT' && npm run format
```
Output:
```
RUN LINT

> linkta@1.0.0 lint
> eslint . && prettier --check .


Core/client/components/layout/MainLayout.tsx
  39:50  warning  Replace `"layout-open"` with `'layout-open'`  prettier/prettier

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

Checking formatting...
[warn] components/layout/MainLayout.tsx
[warn] Code style issues found in the above file. Run Prettier with --write to fix.
npm run lint exited with status code 1
RUN FORMAT

> linkta@1.0.0 format
...
```

However, that's also not the behavior we want to see - I believe `npm run lint` exiting with a non-zero status code should exit the pre-commit hook entirely. 

I think swapping the order of the format and lint commands to `cd client && npm run format && npm run lint`, such that formatting will be applied before linting, is a better solution. However, we may also want to be wary of the `&&` syntax and move these commands to separate lines altogether. 

### 2. Applying changes

While fixing the commands will allow the formatting to be applied to files, any new changes resulting from the linting and formatting scripts will not currently be staged. So those fixes will be left out of the commit. 

I'm unsure if lint-staged would normally handle this issue, but if we intend to use the pre-commit hook to auto-apply formatting, we need to ensure that changes made are staged prior to the hook completing the commit.